### ISSUE TITLE
Receive error "The id_token_hint is invalid" when logging out using angular-oauth2-oidc

### ISSUE THREAD

**Issue by paulspencerwilliams** [FusionAuth/fusionauth-issues#350]
Receive error "The id_token_hint is invalid" when logging out using angular-oauth2-oidc

### Description 
When using angular-oath2-oidc to negotiate OpenId Connect authentication, logout fails with the "If you want the nerdy explanation": 
```
{
  "error" : "invalid_request",
  "error_description" : "The id_token_hint is invalid.",
  "error_reason" : "invalid_id_token_hint"
}
```

### Steps to reproduce

Use Angular code similar to: 
```
const authConfig: AuthConfig = {
    issuer: 'http://localhost:9011',
    redirectUri: window.location.origin + '',
    clientId: "abc-123",
    scope: 'openid profile email offline_access api',
    showDebugInformation: true,
    responseType: 'code'
}

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html'
})
export class AppComponent {
    title = 'app';
    constructor(private oauthService: OAuthService) {
        this.oauthService.configure(authConfig);
        this.oauthService.tokenValidationHandler = new JwksValidationHandler();
        this.oauthService.loadDiscoveryDocument()
          .then(() => {
            this.oauthService.tryLogin()
              .then(() => {
                this.oauthService.loadUserProfile();
                this.oauthService.setupAutomaticSilentRefresh();
              });
          });
    }
}
```

Log in and record access and id tokens (they're shortlived, and only valid on local demo version so I include here for diagnosis):
```
{
  "access_token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZX0.S_6-X3UJ37D8GRTOphhSwnfCqE1Z_4G-SKDNR1n7eH2_wBf5ovchRM-dh5iB5iK1NDasZyPOnw0se6OnXajpZ2dghvS1tBCstFLsSBuUO2tyrb2dTwhmD42txosDwd3kVHo0qBqHOlADZt299YXirS-5ktCMkGwBW6TjiFRRuYjcv7MvL-1jNNPrpmT7cDOAUHqz8fb3Hj6Culf018pqUd9IZWvyUh9um2ZUnxeCI3YqnMuAiLbNmPusGqZMYlqNTf5DvutUvtaHJV7RSwaekFzwmyENsP2BbJHJaFBwqaKCSwz2zYE-vTVoxyR0vHiXBBKvCGs4uNTc1J2UmaIjgA",
  "expires_in": 3599,
  "id_token": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXRfaGFzaCI6IlBQMFRUbTF4WkgtVzdPVjY3VVBaTXciLCJjX2hhc2giOiI2WFloSl93aHdOaHlicl9HNjYtNGt3Iiwibm9uY2UiOiJWd3UwS1cxWFN4YjZhZENEZjR6T2F3cFIwcmF5TnY3Y0V0QkFWWThTTGVNY0sifQ.OfAh4EKWuiUyPUjNnneVKbaGzSOB5E0H2YX1smQvNa_Y43bTZn_Vn4_qo8IfOmgG9y83rt-OLTpBUlqkSSWiWV_JeNFj6gQwfl2xrY19mHG--Nufz1TMc-_fKHh9geh8HVppH-GufvKC0z9XWsXtL8pXJZVq4x6e9_DrdPKoICKsVulnLTdwn6cOL0dyW-E1zTj_1iizYCQbRSvRjs6tlRrNkxRtu_-7sq_9JwKnhlPDBnLDgXh_vH54CIKXy_S9zAbw3Q8gt9F58yhKuaywER7O22qih07bWDW1oAyOa0KPKuGeW__VuZGqX3j6ItbA9MGHuzl-1aoMt7O7Kc_13g",
  "refresh_token": "NJdqU2rGgWHfJt0OJ6gMN9ryd1PSGDIcsKDnf-K-XdhdJfPdHLf_kg",
  "token_type": "Bearer",
  "userId": "c173405e-e9f0-47bc-abfd-bf3586462070"
}
```

Quickly, and thus before token expiration, attempt to logout of FusionAuth using angular-oath2-oidc:
```
oauthService.logOut();
```
which should result in the following GET request: 

```
http://localhost:9011/oauth2/logout?id_token_hint=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXRfaGFzaCI6IlBQMFRUbTF4WkgtVzdPVjY3VVBaTXciLCJjX2hhc2giOiI2WFloSl93aHdOaHlicl9HNjYtNGt3Iiwibm9uY2UiOiJWd3UwS1cxWFN4YjZhZENEZjR6T2F3cFIwcmF5TnY3Y0V0QkFWWThTTGVNY0sifQ.OfAh4EKWuiUyPUjNnneVKbaGzSOB5E0H2YX1smQvNa_Y43bTZn_Vn4_qo8IfOmgG9y83rt-OLTpBUlqkSSWiWV_JeNFj6gQwfl2xrY19mHG--Nufz1TMc-_fKHh9geh8HVppH-GufvKC0z9XWsXtL8pXJZVq4x6e9_DrdPKoICKsVulnLTdwn6cOL0dyW-E1zTj_1iizYCQbRSvRjs6tlRrNkxRtu_-7sq_9JwKnhlPDBnLDgXh_vH54CIKXy_S9zAbw3Q8gt9F58yhKuaywER7O22qih07bWDW1oAyOa0KPKuGeW__VuZGqX3j6ItbA9MGHuzl-1aoMt7O7Kc_13g&post_logout_redirect_uri=http://localhost:61424
```
This, on my instance of FusionAuth results in the above error message and detailed failure cause.

For clarify, here are per line representations of the tokens and url for easy comparision to ensure same values:

```
access_token:  eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZX0.S_6-X3UJ37D8GRTOphhSwnfCqE1Z_4G-SKDNR1n7eH2_wBf5ovchRM-dh5iB5iK1NDasZyPOnw0se6OnXajpZ2dghvS1tBCstFLsSBuUO2tyrb2dTwhmD42txosDwd3kVHo0qBqHOlADZt299YXirS-5ktCMkGwBW6TjiFRRuYjcv7MvL-1jNNPrpmT7cDOAUHqz8fb3Hj6Culf018pqUd9IZWvyUh9um2ZUnxeCI3YqnMuAiLbNmPusGqZMYlqNTf5DvutUvtaHJV7RSwaekFzwmyENsP2BbJHJaFBwqaKCSwz2zYE-vTVoxyR0vHiXBBKvCGs4uNTc1J2UmaIjgA
id_token:      eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXRfaGFzaCI6IlBQMFRUbTF4WkgtVzdPVjY3VVBaTXciLCJjX2hhc2giOiI2WFloSl93aHdOaHlicl9HNjYtNGt3Iiwibm9uY2UiOiJWd3UwS1cxWFN4YjZhZENEZjR6T2F3cFIwcmF5TnY3Y0V0QkFWWThTTGVNY0sifQ.OfAh4EKWuiUyPUjNnneVKbaGzSOB5E0H2YX1smQvNa_Y43bTZn_Vn4_qo8IfOmgG9y83rt-OLTpBUlqkSSWiWV_JeNFj6gQwfl2xrY19mHG--Nufz1TMc-_fKHh9geh8HVppH-GufvKC0z9XWsXtL8pXJZVq4x6e9_DrdPKoICKsVulnLTdwn6cOL0dyW-E1zTj_1iizYCQbRSvRjs6tlRrNkxRtu_-7sq_9JwKnhlPDBnLDgXh_vH54CIKXy_S9zAbw3Q8gt9F58yhKuaywER7O22qih07bWDW1oAyOa0KPKuGeW__VuZGqX3j6ItbA9MGHuzl-1aoMt7O7Kc_13g
id_token_hint: eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkFsZkxLMXU2MGM1dVBEZVdNR1NITDNhQ1BFNCJ9.eyJhdWQiOiIzMTI1ODdmOS0yYWY5LTQ4ODQtYTk5Yi0wMTRjOGMyNDMyNjMiLCJleHAiOjE1NzE5MDU3MjUsImlhdCI6MTU3MTkwMjEyNSwiaXNzIjoiaHR0cDovL2xvY2FsaG9zdDo5MDExIiwic3ViIjoiYzE3MzQwNWUtZTlmMC00N2JjLWFiZmQtYmYzNTg2NDYyMDcwIiwiYXV0aGVudGljYXRpb25UeXBlIjoiUEFTU1dPUkQiLCJlbWFpbCI6InBhdWwud2lsbGlhbXNAZGF0YXBhdGguY28udWsiLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXRfaGFzaCI6IlBQMFRUbTF4WkgtVzdPVjY3VVBaTXciLCJjX2hhc2giOiI2WFloSl93aHdOaHlicl9HNjYtNGt3Iiwibm9uY2UiOiJWd3UwS1cxWFN4YjZhZENEZjR6T2F3cFIwcmF5TnY3Y0V0QkFWWThTTGVNY0sifQ.OfAh4EKWuiUyPUjNnneVKbaGzSOB5E0H2YX1smQvNa_Y43bTZn_Vn4_qo8IfOmgG9y83rt-OLTpBUlqkSSWiWV_JeNFj6gQwfl2xrY19mHG--Nufz1TMc-_fKHh9geh8HVppH-GufvKC0z9XWsXtL8pXJZVq4x6e9_DrdPKoICKsVulnLTdwn6cOL0dyW-E1zTj_1iizYCQbRSvRjs6tlRrNkxRtu_-7sq_9JwKnhlPDBnLDgXh_vH54CIKXy_S9zAbw3Q8gt9F58yhKuaywER7O22qih07bWDW1oAyOa0KPKuGeW__VuZGqX3j6ItbA9MGHuzl-1aoMt7O7Kc_13g
```
### Expected behavior
Logout works, user is redirected to the client app and is presumably unable to refresh tokens, although still able to use the shortlived access token until expiration. 

### Platform
(Please complete the following information)
 - Device: Desktop
 - OS: Windows 10
 - Browser + version Firefox 69

**Comment by robotdan** [comment]
Thanks for the info, here is the JSON body of your `id_token`. 

```

  "aud": "312587f9-2af9-4884-a99b-014c8c243263",
  "exp": 1571905725,
  "iat": 1571902125,
  "iss": "http://localhost:9011",
  "sub": "c173405e-e9f0-47bc-abfd-bf3586462070",
  "authenticationType": "PASSWORD",
  "email": "paul.williams@datapath.co.uk",
  "email_verified": true,
  "at_hash": "PP0TTm1xZH-W7OV67UPZMw",
  "c_hash": "6XYhJ_whwNhybr_G66-4kw",
  "nonce": "Vwu0KW1XSxb6adCDf4zOawpR0rayNv7cEtBAVY8SLeMcK"
}
```

Because the `applicationId` claim is not included in this token it means you are not registered for the application you're attempting to logout of.  Without this claim we cannot resolve the `client_id` and so it is considered an invalid `id_token_hint`.  

For now, ensure that you have a registration for the Application with Id `312587f9-2af9-4884-a99b-014c8c243263` and verify it works as you expect. 

We can revisit this requirement, perhaps there is another way to handle this scenario.

**Comment by robotdan** [comment]
I talked myself through this and I think we should change this behavior, if you logged into SSO (which you can do without being registered), we should be able to log you out with the `id_token`. 

Should be fixed in the next release.  Thanks for reporting!

**Comment by robotdan** [comment]
Available in 1.11.0
